target/linux/armvirt/patches-5.15/221-armvirt-disable-gc_sections.patch

```diff
@@ -0,0 +1,28 @@
+From b77c0ecdc7915e5c5c515da1aa6cfaf6f4eb8351 Mon Sep 17 00:00:00 2001
+From: Mathew McBride <matt@traverse.com.au>
+Date: Wed, 28 Sep 2022 16:39:31 +1000
+Subject: [PATCH] arm: disable code size reduction measures
+ (gc-sections,-f*-sections)
+
+This interferes with the EFI boot stub on armv7l.
+
+Signed-off-by: Mathew McBride <matt@traverse.com.au>
+---
+ arch/arm/Kconfig | 1 -
+ 1 file changed, 1 deletion(-)
+
+diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
+index 66f5d6c35..4ebd51204 100644
+--- a/arch/arm/Kconfig
++++ b/arch/arm/Kconfig
+@@ -117,7 +117,6 @@ config ARM
+ 	select HAVE_UID16
+ 	select HAVE_VIRT_CPU_ACCOUNTING_GEN
+ 	select IRQ_FORCED_THREADING
+-	select HAVE_LD_DEAD_CODE_DATA_ELIMINATION
```

Comment:
Do you know which sections are needed?
We could add a check to keep them for kernels with EFI.

Reply:
This set has more infos:
https://lore.kernel.org/linux-arm-kernel/20181106133935.GB30658@n2100.armlinux.org.uk/
The conclusion there was that gc-sections on arm isn't worth it.

I tried it at one point, and the picture shifted a bit when comparing final compressed images, so there was still a gain. So alternatively we could replace this patch here with that unmerged set there.

On top of this, there's another hint in that thread to squeeze out a little more:
```
From: Andre Heider <a.heider@gmail.com>
Date: Mon, 6 Feb 2023 16:44:09 +0100
Subject: [PATCH] kbuild: sort sections by alignment for
 LD_DEAD_CODE_DATA_ELIMINATION

Due to the higher number of different input sections with
-f{function,data}-section, there may be lots of padding holes, since
each sections has its own alignment.

Placing larger alignments before smaller alignments can reduce the
amount of padding needed.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 028d23841df8..27ffe3362195 100644
--- a/Makefile
+++ b/Makefile
@@ -938,7 +938,7 @@ endif
 
 ifdef CONFIG_LD_DEAD_CODE_DATA_ELIMINATION
 KBUILD_CFLAGS_KERNEL += -ffunction-sections -fdata-sections
-LDFLAGS_vmlinux += --gc-sections
+LDFLAGS_vmlinux += --gc-sections --sort-section=alignment
 endif
 
 ifdef CONFIG_SHADOW_CALL_STACK
-- 
2.39.1
```